eventsourcing/src/main/java/org/axonframework/eventsourcing/annotation/CriteriaResolverDefinition.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventsourcing.annotation;
+
+import jakarta.annotation.Nonnull;
+import org.axonframework.configuration.NewConfiguration;
+import org.axonframework.eventsourcing.CriteriaResolver;
+
+/**
+ * Defines how a {@link CriteriaResolver} should be constructed for an {@link EventSourcedEntity} annotated class. The
+ * definition receives the {@code entityType} and {@code idType} to create the resolver for. In addition, it receives
+ * the {@link NewConfiguration} to resolve any component dependencies that are necessary for creating the resolver.
+ *
+ * @author Mitchell Herrijgers
+ * @since 5.0.0
+ */
+@FunctionalInterface
```

Comment:
If we don't have this annotation on the `EventSourcedEntityFactoryDefinition`, then I don't think we should have it here either.
```suggestion
```

Reply:
You're right, removing